### DISCUSSION TITLE
카카오맵을 사용하는 매크로에 대해 질문 있습니다.

### DISCUSSION THREAD

**Discussion by 20200898** [changdoc/naver-vaccine-macro#152] [Q&A]
해당 매크로를 실행해두면 네이버 예약으로 자동으로 들어가서 예약하나요? 만약 그렇다면 네이버 예약이 켜진 후에는 수동으로 눌러야 하나요?

**Comment by changdoc** [comment]
카카오맵에 노출되는 매크로 동작을 하려면
네이버 로그인 해둔 상태에서
예약할 지역을 네이버 맵에서 graphql로 데이터 추출, 병원 목록 생성
동일한 구역을 카카오맵에서 새로고침 후 left_by_count_coords request body를 추출 후 저장
한 뒤에 동작 시키시면 카카오맵상에서 잔여 백신 수가 확인이 되면 네이버 예약 신청 페이지로 이동 후 예약 시도를 하는 형식으로 동작하는데요.
![image](https://user-images.githubusercontent.com/1879494/127588085-a4296ae6-a3ca-4104-9744-92caef24d1a4.png)

데이터를 넣어야할게 2가지로 동작 방식도 좀 복잡하여 아직 가이드 작성도 못한상태네요.

**Reply by 20200898** [reply]
둘 모두 저장해둔 뒤에 https://github.com/changdoc/naver-vaccine-macro/discussions/90 이 링크의 카카오맵 매크로를 실행하면 되는 것이죠? 감사합니다.

**Reply by changdoc** [reply]
해당 매크로는 독립적으로 설정해서 사용하시는거라 제 매크로 동작과는 관계가 없습니다.

제가 작업한 부분은 해당 매크로 없이 카카오 맵에서 새로고침 수행하며 잔여 백신 조회 하다 발생하면 네이버 예약 신청 페이지로 이동하는식입니다.

**Reply by 20200898** [reply]
아 그리고, 예약할 지역과 동일한 구역이라고 하셨는데 예약할 지역에 카카오맵 매크로에서 새로고침하는 지역이 포함될 경우 상관 없는 것이죠? 완벽히 똑같을 필요가 있나요?

ex) 예약할 지역의 병원이 ( a b c d e ) 라고 할 때  카카오맵 매크로에서 새로고침 하는 지역의 병원이 ( b c d ) 일 경우

**Reply by 20200898** [reply]
> 해당 매크로는 독립적으로 설정해서 사용하시는거라 제 매크로 동작과는 관계가 없습니다.
> 
> 제가 작업한 부분은 해당 매크로 없이 카카오 맵에서 새로고침 수행하며 잔여 백신 조회 하다 발생하면 네이버 예약 신청 페이지로 이동하는식입니다.

아 그런 방식이었군요 감사합니다~! 이해 되었어요!

**Reply by changdoc** [reply]
완벽히 똑같지는 않을 수 있을 듯한데요.
카카오맵 api 결과에는 병원이 있는데 네이버 맵 graphql에서 추출하여 생성한 병원 목록에는 없는 병원에서 만약 잔여 백신이 발생한다면 네이버 예약 신청 페이지로 가는 링크를 생성할 방법이 없어 해당 상황에서는 예약 신청을 못하게 됩니다.
